Packs/SymantecCloudSOC/Integrations/SymantecCloudSOC/SymantecCloudSOC.py

```diff
@@ -0,0 +1,489 @@
+
+
+def add_fields_to_event(event: dict, log_type: str) -> None:
+    """
+    Adds the _time and type keys to the events.
+    Args:
+        events (List[Dict]): A list of events to add the _time and type keys to.
+    Returns:
+        list: The events with the _time and type keys.
+    """
+    if event:
+        if log_type == "Incident_logs":
+            event['_time'] = event.get('incident_start_time')
+            event['type'] = "Detect incident"
+        elif log_type == "Investigate_logs":
+            event['_time'] = event.get('created_timestamp')
+            event['type'] = "Investigate"
+
+
+def dedup_by_id(last_run: dict, events: list, log_type: str, limit: int,
+                number_of_events: int, last_fetch: str) -> Tuple[list, dict]:
+    """
+    Dedup mechanism for the fetch to check both log_id and created_timestamp/incident_start_time
+    (since timestamp can be duplicate)
+    Args:
+        last_run (dict): Last run.
+        events (list): List of the events from the API.
+        log_type (str): the log type.
+        limit (int): The number of events to return.
+        number_of_events (int): The number of event we already fetched
+        last_fetch (str): Last fetch time.
+    Returns:
+        - list of events to send to XSIAM.
+        - The new last_run (dictionary with the relevant timestamps and the events ids).
+        - The new last_run timestamps.
+    """
+    last_run_ids = dict_safe_get(last_run, [f'{log_type}-ids'], default_return_value=[])
+    last_run_time = dict_safe_get(last_run, ["last_run"]) or last_fetch
+    new_events: list = []
+    new_events_ids = []
+    new_last_run: dict = {}
+    new_last_run_time: str = last_run_time
+    new_last_run_time_date = datetime.strptime(new_last_run_time, DATE_FORMAT_SYMANTEC)
+    # The logs sort by asc by default
+    if events:
+        for event in events:
+            if len(new_events) + number_of_events < limit:
+                event_timestamp = (event.get("incident_start_time")
+                                   if log_type == "Incident_logs"
+                                   else event.get("created_timestamp"))
+                event_id = event.get('_id')
+                # The event we are looking at has the same timestamp as previously fetched events
+                if event_timestamp == last_run_time:
+                    if event_id not in set(last_run_ids):
+                        new_events.append(event)
+                        last_run_ids.append(event_id)
+                # The event has a timestamp we have not yet fetched meaning it is a new event
+                else:
+                    new_events.append(event)
+                    new_events_ids.append(event_id)
+                    # If the event has a timestamp newer than the saved one, we will update the last run to the
+                    # current event time
+                    event_timestamp_date = datetime.strptime(event_timestamp, DATE_FORMAT_SYMANTEC)
```

Comment:
You're doing this a lot during the flow.
Better to create a helper function that will get the timestamp you're using and will return it after the casing.
This way you're just doing the `datetime.strptime({event_time}, DATE_FORMAT_SYMANTEC)` once